### ISSUE TITLE
Hover-dismissable: Use cntl rather than esc

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#1393]
Some interesting research from @smhigley:
https://youtu.be/lb0_v7D4kbs?t=1531

Basically: People were worried esc would have other effects, e.g. closing a modal, or other screenreaders. Using the cntl key was more effective, although discoverability is an issue.

We can update the [understanding doc](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html), and [technique](https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR39), and possibly create another technique that incorporates the position of a close button?

**Comment by patrickhlauke** [comment]
discoverability is indeed the killer for this one, so that aspect should probably be aknowledged. one way around the conundrum may well be to listen to both `Esc` *and* `Ctrl` (provided you're handling the mentioned issue in situations like tooltips in modals).

i'd also say that the fear of pressing `Esc` is likely only in those scenarios where users are worried of triggering a cascade of other things closing. For simpler scenarios (which are likely more common), `Esc` is probably still the best option.

**Comment by alastc** [comment]
Fair enough. Still worth an update, just needs assigning.

**Comment by mraccess77** [comment]
We discussed a close button previously - but the close button couldn't be the only thing relied on as it requires either pointer or focus movement.

**Comment by alastc** [comment]
@mraccess77 yep, I wasn't suggesting that as a completely separate approach, but it could be a separate technique. We have a technique at the moment that focuses on the esc key approach and doesn't mention a close button.

**Comment by mraccess77** [comment]
If esc is the issue we might have a new technique that has both a keyboard shortcut  that is not ESC and has a close button.  That makes sense and would solve both concerns.

**Comment by patrickhlauke** [comment]
can't (or at least it would be *very* weird to) have a close button on things like tooltips, menus that appear on hover, etc.

so i'd propose expanding the existing technique to listen to *both* `esc` and `ctrl` to start with

> if (((e.keyCode || e.which) === 27) || ((e.keyCode || e.which) === 17))

**Comment by alastc** [comment]
> it would be very weird to have a close button on things like tooltips, menus that appear on hover, etc.

@patrickhlauke did you watch the video? With "dismissable" people are assuming that "A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus" means that there needs to be one for interfaces without a keyboard, so there needs to be a pointer-accessible close button.

**Comment by patrickhlauke** [comment]
so the strict interpretation of the SC for "but if there's no keyboard" essentially made all current tooltips, popovers, menus illegal unless they have a visible and clickable close button? i'm fairly sure that was not discussed nor advertised as such at the time...

**Comment by patrickhlauke** [comment]
> > it would be very weird to have a close button on things like tooltips, menus that appear on hover, etc.
> 
> @patrickhlauke did you watch the video? With "dismissable" people are assuming that "A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus" means that there needs to be one for interfaces without a keyboard, so there needs to be a pointer-accessible close button.

coming back to this a bit late, but: wouldn't moving the mouse pointer to the close button count as "moving pointer hover"? again, fairly sure that when the SC was written, nobody was thinking of the "pointer, but no keyboard" scenario, and it may not be fully possible to square this scenario with the SC as written (unless it's slightly reinterpreted/clarified that it's ok for the pointer to move "within the tooltip/content" to get to the control. note also that requiring a close button/control for any hover-based things will have a huge impact on all sites that use anything like a tooltip, dropdown menu, popup, etc.
